src/main/java/org/openstreetmap/atlas/checks/event/CheckFlagEvent.java

```diff
@@ -85,17 +132,38 @@ public static JsonObject flagToFeature(final CheckFlag flag,
                                 Optional.ofNullable(flagProperties.getAsJsonPrimitive("name"))
                                         .map(JsonPrimitive::getAsString).orElse("Task"),
                                 decorator)));
-
         // Reference properties lost during GeoJson conversion
         flagProperties.add("feature_properties", featureProperties);
         flagProperties.add("feature_osmids", uniqueFeatureOsmIds);
-        flagProperties.addProperty("feature_count", geometries.size());
+        flagProperties.addProperty("feature_count", featureProperties.size());
 
         feature.addProperty("id", flag.getIdentifier());
         feature.add("properties", flagProperties);
         return feature;
     }
 
+    private static void populateFlaggedRelationPropertiesAndGeometries(
```

Comment:
@Bentleysb @nachtm how do people feel about static functions that manipulate params and do not return anything?

Reply:
Static is fine, but I prefer when an altered copy is returned. That usually make it easier to keep track of the changes, and chain calls.

Reply:
Agreed! As a turnaround, I would split up this helper function into smaller functions with return types.

Reply:
cool cool